privacyidea/lib/error.py

```diff
@@ -55,6 +55,8 @@ class ERROR:
     SERVER = 903
     USER = 904
     PARAMETER = 905
+    FIREBASE_JSON = 1001
```

Comment:
Hm, I'm not sure if we should add new error codes specifically for the Firebase config validation? Maybe we should use some more generic error code like "CONFIG"?

Reply:
I am not sure. 
Imho the idea of the error codes is, that you can find a dedicated place in the code, where the error occurred.
So I really want to have a dedicated firebase error code.
The question is, if we should define these error codes in the firebase modules with its own number range - this would be a general question.

Reply:
Hm, I'm not sure if we have a connection between error codes and code places currently. For example, ``PARAMETER`` is used in the exception class ``ParameterError``, and ``ParameterError`` is raised at a lot of different places.

If I compare the new constants ``FIREBASE_JSON`` and ``FIREBASE_PROJECT`` with the existing constants like ``SERVER``, ``PARAMETER``, the new constants seem a lot more specific to me, because conceptually the firebase provider is just one of multiple SMS providers.

So I'm not strongly opposed to the new error codes, they just seem a bit inconsistent with the existing error codes to me. :)

Reply:
> So I'm not strongly opposed to the new error codes, they just seem a bit inconsistent with the existing error codes to me. :)

This might be due to the fact, that we did not finally rework the error codes completely.

You could argue, that these errors are no "ParameterError"s, sinde ParameterErrors are errors in the API. But these errors, that are raised here, are directly **logical** errors.

So somehow something new, right?
Maybe we should do a meeting or talk. :-)

Reply:
Hi @fredreichbier 
I reworked to use the ConfigAdminError. I think this makes a lot of sense now.

Reply:
Looks good, thx!